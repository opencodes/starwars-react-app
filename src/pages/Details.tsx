import React from 'react'
import { useParams } from 'react-router-dom';
import CharacterProfile from '../components/CharacterProfile'
import { useAppContext } from '../context/AppContext';
import Head from './components/Head';




const Details = () => {
  let { id } = useParams();
  const { characters } = useAppContext();
  const character = characters?.results.find(c => c.id == id)

  // const { data: planets, loading: loadingPlanets, error: planetsErr } = useFetch(
  //     HttpMethods.GET,
  //     END_POINT.planets
  // );
  // const { data: starships, loading: loadingStarships, error: starshipErr } = useFetch(
  //     HttpMethods.GET,
  //     END_POINT.starships
  // );
  // const { data: films, loading: loadingFilms, error: filmErr } = useFetch(
  //     HttpMethods.GET,
  //     END_POINT.films
  // );
  // setFilms(toObject(films.results))
  // setPlanets(toObject(planets.results))
  // setStarships(toObject(starships.results));
  // console.log();
  return (
    <>
      <Head onSearch={() => { }} />
      <div className="container">
        <h5>Profile</h5>
        <hr />
        {character && <CharacterProfile character={character} />}
      </div>
    </>
  )
}

export default Details