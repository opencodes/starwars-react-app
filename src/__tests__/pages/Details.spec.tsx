import Details from '../../pages/Details';
import { renderWithRouter } from '../test-utils';

describe('Details', () => {
  test('should render Page correctly', () => {
    const { asFragment } = renderWithRouter(<Details />);
    expect(asFragment).toMatchSnapshot();
  });
});

