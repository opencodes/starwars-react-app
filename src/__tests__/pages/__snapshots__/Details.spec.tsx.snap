// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details should render Page correctly 1`] = `[Function]`;
